petab/v1/problem.py

```diff
@@ -1185,33 +1185,14 @@ def add_measurement(
         )
 
 
-class VersionNumber(RootModel):
-    root: str | int
-
-
-class ListOfFiles(RootModel):
-    """List of files."""
-
-    root: list[str | AnyUrl] = Field(..., description="List of files.")
-
-    def __iter__(self):
-        return iter(self.root)
-
-    def __len__(self):
-        return len(self.root)
-
-    def __getitem__(self, index):
-        return self.root[index]
-
-
 class SubProblem(BaseModel):
     """A `problems` object in the PEtab problem configuration."""
 
-    sbml_files: ListOfFiles = []
-    measurement_files: ListOfFiles = []
-    condition_files: ListOfFiles = []
-    observable_files: ListOfFiles = []
-    visualization_files: ListOfFiles = []
+    sbml_files: list[str | AnyUrl] = []
```

Comment:
This can be `list[Path | AnyUrl]`, since pydantic has native support for `pathlib.Path` IIRC. But then you might need to make a custom path object that serializes to string https://github.com/dilpath/mkstd/blob/fb4ebfda629b0fb9eedb1def5ab05438ab85696e/mkstd/types/path.py

Fine as is, just noting in case it becomes useful later